### ISSUE TITLE
Issue installing on OSX 10.12

### ISSUE THREAD

**Issue by mkounkel** [gnudatalanguage/gdl#426]
I've attempted to build GDL on my mac in many different ways, all seem to end in a failure. I've tried macports version, to no avail. The default gcc variant is 49, it seems to not play nice with xcode 9. Building with other gcc variants works better, but it still dies in the middle, complaining that

Error: Failed to build gnudatalanguage: command execution failed
Error: See /opt/local/var/macports/logs/_opt_local_var_macports_sources_rsync.macports.org_release_tarballs_ports_math_gnudatalanguage/gnudatalanguage/main.log for details
[main.log](https://github.com/gnudatalanguage/gdl/files/2210917/main.log)

Building from the source, cmake works without a hitch. Make generates a whole slew of warnings along the way (most commonly, warning: control reaches end of non-void function [-Wreturn-type]) before dying with 
4 warnings generated.
6 warnings generated.
10 warnings generated.
make[1]: *** [src/CMakeFiles/gdl.dir/all] Error 2
make: *** [all] Error 2

**Comment by alaingdl** [comment]
*1 reading this log, it seems to be a good idea to deactivate ImageMagick or GraphickMagick
cmake [...] -DGRAPHICSMAGICK=off -DMAGICK=off

*2 did you tried the Brew version ? (after installing Brew)
brew tap brewsci/science
brew install gnudatalanguage

*3 my minismal script shoudl work too

Ways 2 & 3 did not provide current git master version but are useful steps
to be able to compile the current version ! I can help you when your will have one of that ready.

**Comment by mkounkel** [comment]
Well, all of the method suggested have failed also.

1) disabling Image and Graphickmagick did nothing. The make failed in the exact same place - technically it reached 100%, but exits with an error.

2) Brew doesn't like that I have macports, complaining that I have to move one of the system folders, and I would prefer avoid doing that, considering that I use macports much more frequently than I use brew.
The exact place it yells at me is
[  8%] Built target csirocsa
make: *** [all] Error 2

3) The minimal script exits with an error, unable to find the file in a folder it downloaded:
./minimum_script4gdl.sh: line 341: ~/build/cmake-2.8.12/bin/cmake: No such file or directory
make: *** No targets specified and no makefile found.  Stop.
-e 
It seems a problem occured during the compilation of GDL
Please check carrefully the outputs

**Comment by alaingdl** [comment]
1/ using the Git master & Apple LLVM version 8.0.0 (clang-800.0.42.1) & cmake version 3.11.3
I got it with : 
cmake ..  -DOPENMP=OFF -DHDF=OFF -DPYTHON=off -DPSLIB=OFF -DEDITLINE=off -DREADLINE=no

2/ ok, sorry.

3/ hum, sorry again, I have to check the status of this script, it seems that the Git version is not the SVN version ... TBC/TBF

**Comment by alaingdl** [comment]
3/ using https://sourceforge.net/p/gnudatalanguage/svn/HEAD/tree/trunk/gdl/scripts/minimum_script4gdl.sh?format=raw
I succeed to go to the end (with cmake version 3.11.3 & Apple LLVM version 8.0.0 (clang-800.0.42.1))

**Comment by mkounkel** [comment]
And once again that would be a negative to both, same errors as before in their respective cases. Running on cmake version 3.11.4, Apple LLVM version 9.0.0 (clang-900.0.39.2)

**Comment by sflinois** [comment]
You can try using clang-6.0. (I believe it is available on MacPorts under clang-6.0 or llvm-6.0)
I tried compiling with Apple LLVM version 9.1.0 (clang-902.0.39.2) and i just had to turn off OpenMP HDF and PSLIB (I don't have those libs and Apple Clang does not work well with OpenMP)

(Note that the default path using brew is : /usr/local/opt/llvm/bin/)
To compile using clang-6.0 I did :
export CC=/my/path/to/llvm/bin/clang
export CXX=/my/path/to/llvm/bin/clang++
cmake .. -DPSLIB=OFF -DHDF=OFF
make

**Comment by mkounkel** [comment]
I managed to get it working, thanks.

**Comment by ravenep** [comment]
Hi, 

Mac OS woes have come back to haunt the issue board. I was previously using 0.9.7 (compiled from Fink) but need the widget support included with 0.9.8

Have tried unsuccessfully to compile from GitHub (this would be the preferred starting point but at this point will try anything),

Versions of things: 
MacOS 10.12.6
cmake 3.12.0
Apple LLVM version 9.0.0 (clang-900.0.39.2)
plplot: 5.13.0 is installed, but it successfully uses plplot 5.9.9 for gdl install which I understand is critical (PLPLOTDIR=/Users/eraven/prog/plplot-5.9.9)
wxmac (i.e. wxWidgets) 3.0.4 

From @alaingdl 's previous comment, I tried turning off additional options (this failed) and also tried home-brew for sanity (this worked but is the version I already have that does not have option for widgets) 

Then tried @alaingdl 's vanilla gdl install.. this is really great idea having the optimal combination of software versions, but it unfortunately did not work for me. So, some questions:

1. is there an error log buried in there somewhere I could share with the dev team? 
2. what worked for @mkounkel ? 
3. can the @alaingdl vanilla script be updated for GitHub ?

Thanks for any feedback!

**Comment by alaingdl** [comment]
@ravenep Ha ha, I worked last week on my script, an updated version is [here](http://aramis.obspm.fr/~coulais/IDL_et_GDL/minimum_script4gdl.sh). Successfully tested on OSX 10.12 & 10.13 Friday

Not committed on Github just because I had trouble on Fedore28 and also I forgot to turn off GeoTiff 

(and also I hoped to put at the same time a similar script for MSwin from my students)

**Comment by mkounkel** [comment]
I managed to _install_ it with macports changing the clang version, which is 0.9.7 version. I kind of gave up on it after that, since it did not prove functional for my needs.

**Comment by alaingdl** [comment]
@ravenep useful log in 
./CMakeFiles/CMakeOutput.log
./CMakeFiles/CMakeError.log
under the build/ dir. (if any)

**Comment by alaingdl** [comment]
@ravenep ha, I forgot ! I have a bug related to cmake & plplot in my script :cry: 

line 
if [[ ${cmake_version:0:3} < "3.2" ]] ; then
if wrong

I planned to force cmake 3+ but had no time to finish, sorry.

**Comment by alaingdl** [comment]
@ravenep I forgot to mention the most important point : we do have big trouble with widgets on OSX, both locally (in front of the screen) or through the network (ssh -X)

we welcome feedback on this topic !

**Comment by sflinois** [comment]
@ravenep You can also try using the latest Clang version from LLVM (You are currently using Apple Clang, Apple clang have problems with OpenMP and I personally try avoiding it with GDL for now )

You can install it from brew : 
`brew install --with-toolchain llvm`
for Cmake to use it you can change the CC and CXX variables
```
export CC=/usr/local/opt/llvm/bin/clang
export CXX=/usr/local/opt/llvm/bin/clang++
```
Hope that helps ! =)

**Comment by ravenep** [comment]
Thank you @alaingdl @mkounkel @sflinois for your many suggestions. I gave it a few more tries but have given up to get back to an analysis problem & found a hack-y solution using 0.9.7.

I have no software dev experience (which is probably why I failed!) but thought I'd leave some comments / questions as feedback. 

:+1: pre-compiled GDL was the winner. I found home-brew to be an easy install on Mac (fink was tricky but it worked in the end). I also used home-brew to install many of the dependancies. 

:imp: The various combinations of old / new software needed for specific functions (e.g. widgets) is a nightmare. Thank you to those of you that are working on this.

:pray: Has there been any discussion of creating a pre-compiled version of 0.9.8 for Mac? Or, would it be possible to make a custom environment on something like docker? 

I am very happy to provide feedback on widgets if I can get it working - tbd on that. Thanks again to all, and many many thanks to the dev team working on GDL.
